app/src/main/java/org/dhis2/data/forms/dataentry/fields/scan/ScanTextView.kt

```diff
@@ -87,22 +84,12 @@ class ScanTextView @JvmOverloads constructor(
         }
     }
 
-    private fun checkCameraPermission() {
-        if (
-            ContextCompat
-                .checkSelfPermission(context, Manifest.permission.CAMERA) == PERMISSION_GRANTED
-        ) {
-            subscribe()
-            val intent = Intent(context, ScanActivity::class.java)
-            intent.putExtra(Constants.OPTION_SET, optionSet)
-            intent.putExtra(Constants.SCAN_RENDERING_TYPE, renderingType)
-        } else {
-            Toast.makeText(
-                context,
-                R.string.camera_permission_denied,
-                Toast.LENGTH_LONG
-            ).show()
-        }
+    private fun goToScanActivity() {
+        subscribe()
+        val intent = Intent(context, ScanActivity::class.java)
+        intent.putExtra(Constants.OPTION_SET, optionSet)
+        intent.putExtra(Constants.SCAN_RENDERING_TYPE, renderingType)
```

Comment:
I think you have an activity extension that can do this.

Reply:
There isn't an extension for a .startActivityForResult() though.

Reply:
If you would like to do a new one for activityForResult, if not thats fine :+1: . I think you only need to pass the RQ_QR_SCANNER and swap the "startActivity" to "StartActivityWithResult)